streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -268,9 +274,13 @@ public void onPartitionsAssigned(final Collection<TopicPartition> assignment) {
             final long start = time.milliseconds();
             try {
                 if (streamThread.setState(State.PARTITIONS_ASSIGNED) == null) {
-                    return;
```

Comment:
Why did you remove the `return`? Without the `return`, the code will reach the `finally` block and log that the assignment took place.

Reply:
I'm confused, we will reach `finally` block no matter we have `return` or or not right?

Reply:
Oh sorry, I misunderstood the code.